### PR TITLE
Feature/858

### LINKED ISSUE #858: As a guardian, I would like to delete a Citizen, Trustee or Guardian, so that once they are no longer related to the institution, their data can be removed.
## Description
**Describe the Task** \
A Guardian should be able to delete Citizens, Trustees or other Guardians once they leave the institution. Information about the users should also be deleted from the database. 

**Possible Suggested Solution**  
- Be able to delete users from the "users view" when logged in as a Guardian. See figure below with possible option delete other users when logged in as Guardian.  
 

Image

- Expand issue as it's explored. 

## Sub Tasks
- [ ] Catch possible exceptions which could occur
- [ ] Write some unit tests
- [ ] write one or two integration tests
- [ ] If three above substasks are not relevant to your issue explain why 

Related to #625

### PR DESCRIPTION
**Pull request by Obeline**
# Description

Use api-client feature 858 issue/pull request with this.

Now you can delete a user, from their settingstab
When you click the delete button, a dialog appears which asks you to enter the users name, in order to delete them
This is kinda like deleting a github repository, and it is there to ensure no user is accidentally deleted
If you type in the wrong name it will tell you

When you have deleted the user it redirects you back to the select user screen

There is one issue with the code, and that is that it does not update the screen when you delete the last citizen. It is deleted, but for it to dissapear you have to either log out and log in, or add a new user. I don't know how to fix this, i think it might be because of some UserHasNoCitizen exception. 

sletBrugerPrompt

Fixes #858 

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Unit tests, and also just manually deleting and looking in the database.

**Development Configuration**
* Flutter version: 3.3.9
* Dart version: 2.18.5 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [ ] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: test/screens/settings_screens/settings_screen_test.dart [hunk 1]
```diff
@@ -62,14 +66,17 @@ class MockUserApi extends Mock implements UserApi {
 
 void main() {
   Api api;
+  MockAccountApi accountApi;
   SettingsBloc settingsBloc;
 
-  final DisplayNameModel user = DisplayNameModel(
```
Comment (NikolajKrogh) [inline test/screens/settings_screens/settings_screen_test.dart:67]: Hvorfor har fjernet final?

Comment (Obeline) [reply test/screens/settings_screens/settings_screen_test.dart:67]: Den bliver brugt i den nederste, lidt cursed test

**Review by NikolajKrogh — APPROVED** [review]
lgtm

**Review by manfredhod — APPROVED** [review]
LGTM

**Pull request merged**